lib/plane.rb

```diff
@@ -0,0 +1,26 @@
+require_relative './airport'
+
+class Plane
+
+  def initialize
+    @current_airport = nil
+    @flying = true
+  end
+
+  def land(airport)
+    raise 'error, plane is not flying' if @flying == false
+    raise 'airport full' if airport.hanger.count == airport.DEFAULT_CAPACITY
+    raise 'landing denied, there is a storm!' if airport.storm == true
+    @current_airport = airport
+    @flying = false
+    airport.hanger << self
+  end
+
+  def take_off(airport)
+    raise 'plane is not in the specified airport' if airport != @current_airport
+    raise 'take-off denied, there is a storm!' if airport.storm == true
+    @flying = true
+    airport.hanger.reject! { |plane| plane == self }
```

Comment:
```suggestion
    airport.hanger.delete(self) # this would return the deleted plane
    airport.hanger - [self] # this would return the array, without the deleted plane
```
I'm not 100% sure these would work with your code, I'm only suggesting because this line took me a little while to figure out, so just trying to maybe suggest something that improves readability - maybe you tried these suggestions and they didn't work, not trying to teach you to suck eggs!